File: client/src/app/_models/category.ts
// Add this interface extension to your category model or create a new interface
// client/src/app/_models/category.ts - ADD TO EXISTING FILE

export interface Category {
  id: number;
  name: string;
  description?: string;
  iconName?: string;
  colorCode?: string;
  parentCategoryId?: number;
  subCategories?: Category[];
  expanded?: boolean;

  // Add these properties for home page display
  displayIcon?: string;
  displayColor?: string;
}

File: API/Entities/Category.cs
namespace API.Entities;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string IconName { get; set; } = string.Empty;
    public string ColorCode { get; set; } = string.Empty;
    public int? ParentCategoryId { get; set; }
    public Category? ParentCategory { get; set; }
    public ICollection<PredictionCategory> PredictionCategories { get; set; } = [];
    public ICollection<Category> SubCategories { get; set; } = [];
}